File: Practica 5/Ecuaciones_Cuadraticas/POO.java
package Ecuaciones_Cuadraticas;
import java.util.Scanner;

public class POO {
    private double a;
    private double b;
    private double c;

    public POO(double a, double b, double c) {
        this.a = a;
        this.b = b;
        this.c = c;
    }
    public double getDiscriminante(){
        return b*b - 4*a*c;
    }
    public double getRaiz1(){
        return (-b + Math.sqrt(getDiscriminante()))/(2*a);
    } 
    public double getRaiz2(){
        return (-b - Math.sqrt(getDiscriminante()))/(2*a);
    } 
    public void resolver(){
        double d=getDiscriminante();
        if(d>0){
            System.out.printf("La ecuacion tiene dos raices: %.6f y %.6f%n", getRaiz1(), getRaiz2());
        }
        else if(d==0){
            System.out.printf("La ecuacion tiene una raiz: %.6f%n ",getRaiz1());
        }
        else {
            System.out.println("La ecuacion no tiene raices reales");
        }
    }
    public static void main(String[] args) {
        Scanner scanner=new Scanner(System.in);
        
        System.out.print("Ingrese a, b, c: ");
        double a=scanner.nextDouble();
        double b=scanner.nextDouble();
        double c=scanner.nextDouble();
        
        POO e1=new POO(a,b,c);
        e1.resolver();
    }
    
}

File: Practica 5/Ecuaciones_Cuadraticas/POO.py
import math

class Ecuaciones:
    def __init__(self, a, b, c):
        self.a = a
        self.b = b
        self.c = c

    def getDiscriminante(self):
        return self.b * self.b - 4 * self.a * self.c

    def getRaiz1(self):
        return (-self.b + math.sqrt(self.getDiscriminante())) / (2 * self.a)

    def getRaiz2(self):
        return (-self.b - math.sqrt(self.getDiscriminante())) / (2 * self.a)

    def resolver(self):
        d = self.getDiscriminante()
        if d > 0:
            print("La ecuacion tiene dos raices: %.6f y %.6f" % (self.getRaiz1(), self.getRaiz2()))
        elif d == 0:
            print("La ecuacion tiene una raiz: %.6f" % self.getRaiz1())
        else:
            print("La ecuacion no tiene raices reales")

def main():
    print("Ingrese a, b, c: ")
    entrada = input()
    a, b, c = map(float, entrada.split())

    e1 = Ecuaciones(a, b, c)
    e1.resolver()

if __name__ == "__main__":
    main()
